c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -884,6 +885,11 @@ class DeviceCachingAllocator {
   bool set_fraction = false;
 
   bool record_history = false;
+
+  // enable record trace when any trace tracker has been attached to the
+  // allocator
+  bool enable_trace_tracker_ = false;
```

Comment:
This can just be replaced with `trace_trackers_.size() > 0`, same number of instructions to check.